e2e_playwright/st_spinner_test.py

```diff
@@ -18,6 +18,22 @@
 
 
 def test_spinner_execution(app: Page):
-    app.get_by_test_id("stButton").locator("button").click()
+    app.get_by_test_id("stButton").locator("button").nth(0).click()
     expect(app.get_by_test_id("stSpinner")).to_have_text("Loading...")
     check_top_level_class(app, "stSpinner")
+
+
+def test_spinner_elapsed_time(app: Page):
+    app.get_by_test_id("stButton").locator("button").nth(1).click()
+    expect(app.get_by_test_id("stSpinner")).to_contain_text("Loading...")
+
+    # To not make this flaky, we don't check that the timer shows exactly 0.0 seconds,
+    # but just that it's somewhere around 0.x seconds
+    expect(app.get_by_test_id("stSpinner")).to_contain_text("0.")
+    expect(app.get_by_test_id("stSpinner")).to_contain_text("seconds")
+
+    # Similarly, we wait for 1 second and check that it shows 1.x seconds.
+    # TODO: Need to check in practice if this is flaky.
+    expect(app.get_by_test_id("stSpinner")).to_contain_text("1.", timeout=1000)
```

Comment:
Not sure if the timeout here is super clean or if there's a better alternative? Or should we just leave this out because it could create flakiness?